Apps/Contoso.Forms.Puppet/Contoso.Forms.Puppet/ModulePages/OthersContentPage.xaml.cs

```diff
@@ -181,6 +188,46 @@ async void RunMBaaSAsync(object sender, EventArgs e)
         void SignOut(object sender, EventArgs e)
         {
             Auth.SignOut();
+            SignInInformationButton.Text = "User Not Authenticated";
+        }
+
+        private static byte[] Base64UrlDecode(string input)
+        {
+            var output = input;
+            output = output.Replace('-', '+'); // 62nd char of encoding
+            output = output.Replace('_', '/'); // 63rd char of encoding
+            switch (output.Length % 4) // Pad with trailing '='s
+            {
+                case 0: break; // No pad chars in this case
+                case 2: output += "=="; break; // Two pad chars
+                case 3: output += "="; break; // One pad char
+                default: throw new System.Exception("Illegal base64url string!");
+            }
+            var converted = Convert.FromBase64String(output); // Standard base64 decoder
+            return converted;
+        }
+
+        public static string Decode(string token, string key)
+        {
+            var parts = token.Split('.');
+            var header = parts[0];
+            var payload = parts[1];
+            byte[] crypto = Base64UrlDecode(parts[2]);
+            byte[] headerData = Base64UrlDecode(header);
+            var headerJson = Encoding.UTF8.GetString(headerData,0, headerData.Length);
+            byte[] payloadData = Base64UrlDecode(payload);
+            var payloadJson = Encoding.UTF8.GetString(payloadData, 0, payloadData.Length);
+            return payloadJson;
+        }
+
+        async void SignInInformation(object sender, EventArgs e)
+        {
+            if (userInfo!=null)
+            {
+                string accessToken = Decode(userInfo.AccessToken, "");
```

Comment:
By the way I may be wrong, but I got the impression that token can be null and it might express "unauthenticated" user.